cranelift/codegen/src/ir/function.rs

```diff
@@ -495,3 +495,16 @@ impl<'a> Iterator for InstOffsetIter<'a> {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn new_function() {
+        let func = Function::new();
+        let block = func.dfg.make_block();
+        assert!(func.is_leaf());
+        assert_eq!(func.is_block_basic(block), Ok(()));
```

Comment:
This function won't pass the verifier as there is no entry block. The first block for which an instruction is inserted is in principle the entry block. All blocks need to end with a terminator instruction.